db.go

```diff
@@ -1,14 +1,26 @@
 package factory
 
-import "database/sql"
+import (
+	"database/sql"
+	"reflect"
+)
 
 var dbConnection *sql.DB
+var dbDriverName string
 
-// SetDB sets database connection for factory
+// SetDB sets database connection and a driver name for factory
 func SetDB(db *sql.DB) {
 	dbConnection = db
+	// setting a driver name
+	dr := dbConnection.Driver()
+	dv := reflect.ValueOf(dr)
+	dbDriverName = dv.Type().String()
```

Comment:
Moreover, `dbDriverName` (`dv.Type().String()`) is the variable type name. It make `dbDriverName` unstable because it depends the implementations of the specific drivers.

Reply:
Moving `dbDriverName ` will make testing and mocking are inconvenient.

I agree, `dv.Type().String()` it's a not best solution, but what's options?